spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectorConf.scala

```diff
@@ -60,26 +60,37 @@ object CassandraConnectorConf extends Logging {
     }
   }
 
-  def apply(conf: SparkConf): CassandraConnectorConf = {
-    val hostsStr = conf.get(CassandraConnectionHostProperty, InetAddress.getLocalHost.getHostAddress)
+  def processProperty(property: String, cluster: Option[String] = None): String = {
```

Comment:
If this is now going to get called all over the code base, which I see it is, let's simplify it to 'convert' vs processProperty.
What if the property is null or has length 0? THere's no validation.